build-setup/linux.md

```diff
@@ -237,6 +237,67 @@ sudo make install
 
 to complete the build.
 
+### CMake
+
+CMake version 3.5 is the minimum required to build PyTorch. Download version 3.19.3 from <https://github.com/Kitware/CMake/releases/download/v3.19.3/cmake-3.19.3-Linux-x86_64.sh> and install:
+
+```
+chmod +x cmake-3.19.3-Linux-x86_64.sh
+mkdir /usr/local/cmake
+./cmake-3.19.3-Linux-x86_64.sh --skip-license --prefix=/usr/local/cmake
+```  
+
+In your `.bashrc` file add CMake to `PATH`:
+
+```
+PATH=$PATH:/usr/local/cmake/bin
+```
+
+### Python 
+
+PyTorch currently requires Python 3.6, 3.7 or 3.8, and version 3.7 appears to cause fewest problems in their test status matrix, so we use that. If your system does not have a requisite version of Python install it with a package manager or build the last 3.7 release from source by downloading `Python-3.7.9.tgz` from <https://www.python.org/ftp/python/3.7.9/Python-3.7.9.tgz> then extract and build:
+
+```
+tar -xzf Python-3.7.9.tgz
+cd Python-3.7.9/
+./configure --enable-optimizations
+make altinstall
```

Comment:
Does this install to a directory that requires root access?  If so it needs `sudo`, but for all the other packages we build from source we do the bulk of the build as non-root and just the install as root, i.e. this sequence occurs multiple times in this file:

```
make
sudo make install
```

Maybe there's an equivalent for Python like:

```
make
sudo make altinstall
```

Or if not then probably the whole build needs to be under `sudo`.